File: 2st/Riyeon/전국 대회 선발 고사.java
import java.util.*;
class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        int[] ar = new int[rank.length];
        for(int i=0; i<rank.length; i++)
            ar[i] = rank[i];
        Arrays.sort(ar);
        
        int answer = 0, n = 10000;
        for(int i=0; i<ar.length; i++){
            for(int j=0; j<rank.length; j++){
                if(rank[j] == ar[i] && attendance[j] == true){
                    answer += j*n;
                    n /= 100;
                }
            }
        }
        return answer;
    }
}

File: 2st/Riyeon/전국 대회 선발 고사.py
def solution(rank, attendance):
    answer = 0
    li = []
    for i in range(len(rank)):
        if attendance[i] == True:
            li.append(rank[i])
    li.sort()
    n = 1
    for i in [2, 1, 0]:
        for r in range(len(rank)):
            if li[i] == rank[r]:
                answer += n*r
        n*=100
    return answer